rust/src/detect/transforms/xor.rs

```diff
@@ -60,13 +60,17 @@ unsafe fn xor_parse(raw: *const std::os::raw::c_char) -> *mut c_void {
 
 #[no_mangle]
 unsafe extern "C" fn xor_setup(
-    _de: *mut c_void, s: *mut c_void, opt_str: *const std::os::raw::c_char,
+    de: *mut c_void, s: *mut c_void, opt_str: *const std::os::raw::c_char,
 ) -> c_int {
     let ctx = xor_parse(opt_str);
     if ctx.is_null() {
         return -1;
     }
-    return DetectSignatureAddTransform(s, G_TRANSFORM_XOR_ID, ctx);
+    let r = DetectSignatureAddTransform(s, G_TRANSFORM_XOR_ID, ctx);
+    if r != 0 {
+        xor_free(de, ctx);
```

Comment:
Happens if you do not have content to apply the transform to for example

Reply:
For me even ASAN does not flag it. I cannot reproduce it if I do not have a content to apply to the transform to..
How did you find it?

Reply:
Maybe you need to enable it with export `ASAN_OPTIONS="detect_leaks=1"` ?

I got it with something like `alert ip any any -> any any ( xor: "12"; sid:1)` where there is no buffer to do the transform on